packages/components/src/components/flyout-navigation/flyout-navigation/flyout-navigation-utils.ts

```diff
@@ -0,0 +1,40 @@
+import { consoleError, getTagNameWithoutPrefix } from '../../../utils';
+import { type FlyoutNavigationItemInternalHTMLProps } from '../flyout-navigation-item/flyout-navigation-item-utils';
+import { forceUpdate } from '@stencil/core';
+import { type Class, type Theme } from '../../../types';
+
+export const FLYOUT_NAVIGATION_ARIA_ATTRIBUTES = ['aria-label'] as const;
+export type FlyoutNavigationAriaAttribute = (typeof FLYOUT_NAVIGATION_ARIA_ATTRIBUTES)[number];
+
+export const INTERNAL_UPDATE_EVENT_NAME = 'internalUpdate';
+
+export type FlyoutNavigationUpdate = {
+  activeIdentifier: string;
+};
+export type FlyoutNavigationUpdateEvent = FlyoutNavigationUpdate; // to have consistent event types
+
+export const syncFlyoutNavigationItemsProps = (
+  items: HTMLPFlyoutNavigationItemElement[],
+  activeIdentifier: string,
+  theme: Theme
+): void => {
+  items.forEach((item: HTMLPFlyoutNavigationItemElement & FlyoutNavigationItemInternalHTMLProps) => {
+    item.theme = theme;
+    item.open = item.identifier === activeIdentifier;
+    forceUpdate(item);
+  });
+};
+
+export const validateActiveIdentifier = <T extends Class<any>>(
+  instance: InstanceType<T>,
+  items: HTMLPFlyoutNavigationItemElement[],
+  activeIdentifier: string | undefined
+): void => {
+  if (!(activeIdentifier === undefined || !!items.filter((item) => item.identifier === activeIdentifier).length)) {
```

Comment:
`!!items.filter((item) => item.identifier === activeIdentifier).length)`
Does this only validate if there is exactly one matching identifier? What if the same identifier is used multiple times?

Reply:
I added it to #2928